MaloWBotLK/MaloWBotLK.lua

```diff
@@ -83,6 +83,22 @@ function mb_OnEvent(self, event, arg1, arg2, arg3, arg4, ...)
 		if arg1 == "player" and mb_registeredInterruptSpells ~= nil then
 			mb_HandleTargetSpellcast()
 		end
+	elseif event == "UNIT_SPELLCAST_FAILED" then
+		if arg1 == "player" and (arg2 == "Shred" or arg2 == "Backstab") then
+			mb_lastStrafe = mb_time
+			if mb_lastStrafe + 0.5 > mb_time then
```

Comment:
This code seems quite dangerous, if a strafe is started there are multiple scenarios where the strafe won't ever be cancelled. I would rework this logic to start a strafe using StrafeRightStart and at the same time setting an mb_shouldStopStrafingAt variable to mb_time + 0.1 when it's detected that you're not behind your target, and then in mb_HandleAutomaticMovement I would check mb_shouldStopStrafingAt compared to mb_time and stop the strafe if that 0.1 sec has passed, similar to how mb_shouldStopMovingForwardAt currently works.

Reply:
eee65c9 : Now uses "You must be behind your target. " UI_ERROR message in order to start strafing.